### PR TITLE
Parse cassandra batch messages

### PR DESCRIPTION
**Pull request by rukai**
Takes the same approach as CassandraOperation::Result, recreate a bunch of structures so that we can replace the string query with the query AST.
batch queries are now parsed by sql_parser (soon to be replaced with claudes parser)

### CODE REVIEW

File: shotover-proxy/src/frame/cassandra.rs [hunk 1]
```diff
@@ -432,3 +481,24 @@ pub enum CassandraResult {
...
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum BatchStatementTy {
+    Statement(CQL),
+    PreparedId(CBytesShort),
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub struct BatchStatement {
+    ty: BatchStatementTy,
```
Comment (conorbros) [inline shotover-proxy/src/frame/cassandra.rs:493]: Can we just say `type`?

Comment (rukai) [reply shotover-proxy/src/frame/cassandra.rs:493]: nope, `type` is a keyword

Comment (rukai) [reply shotover-proxy/src/frame/cassandra.rs:493]: ahhh I should do BatchStatementTy -> BatchStatementType though

File: shotover-proxy/src/frame/cassandra.rs [hunk 2]
```diff
@@ -93,7 +95,9 @@ impl CassandraFrame {
     // Count the amount of queries in this `CassandraFrame`, this will either be the count of all queries in a BATCH statement or 1 for all other queries
     pub(crate) fn get_message_count(&self) -> Result<NonZeroU32> {
         Ok(match &self.operation {
-            CassandraOperation::Batch(bytes) => get_batch_len(bytes)?,
+            CassandraOperation::Batch(batch) => {
+                NonZeroU32::new(batch.queries.len() as u32).unwrap_or(nonzero!(1u32))
```
Comment (conorbros) [inline shotover-proxy/src/frame/cassandra.rs:99]: Similar to the bug-fix PR, can we add a comment explaining this? i.e. BATCH statements query length can be 0 and for our purposes counting as 1 is fine.

**Review by conorbros — APPROVED** [review]
Just some some things. Other than that, LGTM!

File: shotover-proxy/src/frame/cassandra.rs [hunk 3]
```diff
@@ -110,7 +115,7 @@ impl CassandraFrame {
                         params: body.query_params,
                     }
                 } else {
-                    unreachable!()
+                    unreachable!("We already know the operation is a query")
```
Comment (benbromhead) [inline shotover-proxy/src/frame/cassandra.rs:118]: we can do this in a different PR, but I feel like all the nested `if let else` statements in each match arm could be brought up a level to the match it self:

```rust
let operation = match (frame.opcode, frame.request_body()?) {
            (Opcode::Query, RequestBody::Query(body)) => {}
...
}
```

Then put the unreachable in the fall through arm.

Comment (benbromhead) [inline shotover-proxy/src/frame/cassandra.rs:118]: we can do this in a different PR, but I feel like all the nested `if let else` statements in each match arm could be brought up a level to the match it self:

```rust
let operation = match (frame.opcode, frame.request_body()?) {
            (Opcode::Query, RequestBody::Query(body)) => {}
...
}
```

Then put the unreachable in the fall through arm.

**Review by benbromhead — APPROVED** [review]

File: shotover-proxy/src/frame/cassandra.rs (hunk 3 above)
Comment (rukai) [reply shotover-proxy/src/frame/cassandra.rs:118]: Unfortunately the entire reason we have to do this is because cassandra-protocol splits the logic into separate request_body and response_body methods.
I tried to unify them before but got resistance from the maintainer.
Its something to look into some day but I considered it low priority.
